src/Http/Batch.php

```diff
@@ -207,7 +207,7 @@ private function parseRawHeaders($rawHeaders)
         list($header, $value) = explode(': ', $headerLine, 2);
         $header = strtolower($header);
         if (isset($headers[$header])) {
-          $headers[$header] .= "\n" . $value;
+          $headers[$header] .= ", " . $value;
```

Comment:
Hi @pimjansen I can do something like this, it will work, but maybe it's overkill, do you know how to simulate a return that might contain unwanted headers?

Array method:
```php
$headers[$header] = array_merge(is_array($headers[$header]) ? $headers[$header] : [$headers[$header]], [$value]);
```

Output:
```
array(1) {
  [0]=>
  array(2) {
    ["vary"]=>
    array(3) {
      [0]=>
      string(6) "Origin"
      [1]=>
      string(8) "X-Origin"
      [2]=>
      string(7) "Referer"
    }
    ["content-type"]=>
    string(31) "application/json; charset=UTF-8"
  }
}

```

Concatenation:
```php
$headers[$header] .= ", " . $value;
```

```
array(1) {
  [0]=>
  array(2) {
    ["vary"]=>
    string(25) "Origin, X-Origin, Referer"
    ["content-type"]=>
    string(31) "application/json; charset=UTF-8"
  }
}
```

Reply:
If in correct the header one is the correct one though.

For example when i return 2 content type headers due whatever reason it will concat where it is invalid if im correct?

Reply:
Just checking in from my phone, thus didn't look at all the details. Passing a single element array should behave identically to passing a string. The PSR-7 getHeader method will always return an array anyway and getHeaderLine the comma concatenated array.

I recommend to always pass an array to the Response class, it is never wrong.

Reply:
Yes, I understand what you mean, but this code only parses the responses of batches sent to google api, and in tests google only returns 3 vary and 1 content-type.

Reply:
I don't have any deeper interest in the Google API client, I don't use it myself, so ultimately I don't particularly care about whatever is done here.

That said: I don't think it's a particularly strong argument to implement a fix that is known to break in some cases, just because these cases are *currently* impossible to trigger. The array version preserves the semantics of the message correctly in 100% of the cases, whereas the comma-concatenated version does not. The array version is not really any harder to implement either, it should just be a `[]=` instead of `.=`.